find_api_page fails in py3.x

The `astropy.utils.misc.find_api_page` function's test (`astropy.utils.tests.test_misc.test_api_lookup`) seems to be failing in py 3.x. I suspect this has been long-missed because it's a ``remote-data`` test and only fails in py3.x.

The problem seems to be that ``http.client.HTTPResponse`` (what `urlopen` returns in py3.x) doesn't behave well with `readline`.  If you do a `readline` followed by `read`, you get a `IncompleteRead` error.  Not sure if this is a feature or a bug, but it's pretty straightforward to patch it, so I'll do that shortly.

I can reproduce it.
```python
________________________ test_api_lookup ____________________

    @remote_data
    def test_api_lookup():
>       strurl = misc.find_api_page('astropy.utils.misc', 'dev', False, timeout=3)

anaconda/envs/py3k/lib/python3.5/site-packages/astropy/utils/tests/test_misc.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
anaconda/envs/py3k/lib/python3.5/site-packages/astropy/utils/misc.py:242: in find_api_page
    oistr = uf.read()
anaconda/envs/py3k/lib/python3.5/http/client.py:446: in read
    s = self._safe_read(self.length)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <http.client.HTTPResponse object at 0x7fc1fd144470>, amt = 127

    def _safe_read(self, amt):
        """Read the number of bytes requested, compensating for partial reads.
    
            Normally, we have a blocking socket, but a read() can be interrupted
            by a signal (resulting in a partial read).
    
            Note that we cannot distinguish between EOF and an interrupt when zero
            bytes have been read. IncompleteRead() will be raised in this
            situation.
    
            This function should be used when <amt> bytes "should" be present for
            reading. If the bytes are truly not available (due to EOF), then the
            IncompleteRead exception can be used to detect the problem.
            """
        s = []
        while amt > 0:
            chunk = self.fp.read(min(amt, MAXAMOUNT))
            if not chunk:
>               raise IncompleteRead(b''.join(s), amt)
E               http.client.IncompleteRead: IncompleteRead(52130 bytes read, 127 more expected)

anaconda/envs/py3k/lib/python3.5/http/client.py:594: IncompleteRead
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -905,6 +905,8 @@ Bug Fixes
 
 - ``astropy.utils``
 
+  - Fixed ``find_api_page`` to work correctly on python 3.x [#4378]
+
 - ``astropy.vo``
 
 - ``astropy.wcs``

--- a/astropy/utils/misc.py
+++ b/astropy/utils/misc.py
@@ -233,21 +233,33 @@ def find_api_page(obj, version=None, openinbrowser=True, timeout=None):
         uf = urllib.request.urlopen(baseurl + 'objects.inv', timeout=timeout)
 
     try:
-        # we read these lines so that `oistr` only gets the compressed
-        # contents, not the header information
-        isvers = uf.readline().rstrip().decode('utf-8')  # intersphinx version line
-        proj = uf.readline().rstrip().decode('utf-8')  # project name
-        vers = uf.readline().rstrip().decode('utf-8')  # project version
-        uf.readline().rstrip().decode('utf-8')
-        oistr = uf.read()
+        oiread = uf.read()
+
+        # need to first read/remove the first four lines, which have info before
+        # the compressed section with the actual object inventory
+        idx = -1
+        headerlines = []
+        for _ in range(4):
+            oldidx = idx
+            idx = oiread.index(b'\n', oldidx + 1)
+            headerlines.append(oiread[(oldidx+1):idx].decode('utf-8'))
+
+        # intersphinx version line, project name, and project version
+        ivers, proj, vers, compr  = headerlines
+        if not 'The remainder of this file is compressed using zlib' in compr:
+            raise ValueError('The file downloaded from {0} does not seem to be'
+                             'the usal Sphinx objects.inv format.  Maybe it '
+                             'has changed?'.format(baseurl + 'objects.inv'))
+
+        compressed = oiread[(idx+1):]
     finally:
         uf.close()
 
-    oistr = decompress(oistr).decode('utf-8')
+    decompressed = decompress(compressed).decode('utf-8')
 
     resurl = None
 
-    for l in oistr.strip().splitlines():
+    for l in decompressed.strip().splitlines():
         ls = l.split()
         name = ls[0]
         loc = ls[3]

